### PR TITLE
radvd: cleanups

### PR DESCRIPTION
**Pull request by robbat2**
autotools: improve handling when libcheck is absent
Fix -Wstringop-truncation warning: s/strncpy/strlcpy/
test: cleanup testcases for -Wfree-nonheap-object
gitignore: skip newer build/autoconf artifacts & patch droppings

Signed-off-by: Robin H. Johnson <robbat2@gentoo.org>

### CODE REVIEW

File: gram.y [hunk 1]
```diff
@@ -218,8 +218,8 @@ ifacehead	: T_INTERFACE name
 			}
 
 			iface_init_defaults(iface);
-			strncpy(iface->props.name, $2, IFNAMSIZ-1);
-			iface->props.name[IFNAMSIZ-1] = '\0';
+			memset(&iface->props.name, 0, sizeof(iface->props.name));
```
Comment (stappersg) [inline gram.y:221]: FWIW: _"looks strange"_

_"Strange"_ as in none of the previous _hugs_ did add `memset()`.

File: gram.y [hunk 2]
```diff
@@ -703,8 +703,8 @@ prefixparms	: T_AdvOnLink SWITCH ';'
 #else
 			if (prefix) {
 				dlog(LOG_DEBUG, 4, "using interface %s for 6to4 prefixes on interface %s", $2, iface->props.name);
-				strncpy(prefix->if6to4, $2, IFNAMSIZ-1);
-				prefix->if6to4[IFNAMSIZ-1] = '\0';
+				memset(&prefix->if6to4, 0, sizeof(prefix->if6to4));
```
Comment (stappersg) [inline gram.y:706]: Meanwhile _"less strange"_    (context is my above comment).

File: privsep-linux.c [hunk 3]
```diff
@@ -163,7 +166,8 @@ int privsep_interface_retranstimer(const char *iface, uint32_t rettimer)
 {
 	struct privsep_command cmd;
 	cmd.type = SET_INTERFACE_RETRANSTIMER;
-	strncpy(cmd.iface, iface, sizeof(cmd.iface));
+	memset(&cmd.iface, 0, sizeof(cmd.iface));
```
Comment (stappersg) [inline privsep-linux.c:169]: `;-)`

File: redhat/systemd/radvd.spec.in [hunk 4]
```diff
@@ -17,6 +17,7 @@ BuildRequires: flex-static
 BuildRequires: pkgconfig
 BuildRequires: check-devel
 BuildRequires: systemd
+BuildRequires: libbsd
```
Comment (stappersg) [inline redhat/systemd/radvd.spec.in:20]: ( Debian maintainer makes a mental note about a potional extra _build dependency_.)

**Comment by stappersg** [comment]
I would like to see that the _commit message_ says something about
the `s/strncpy()/strlcpy()/` change.

File: gram.y (hunk 1 above)
Comment (robbat2) [reply gram.y:221]: I'm not 100% sure the `memset` is required, except in cases where the variable is re-used, and we might leak part of what the old content used to be.
```
char s[16] = "longstring";
// s actually contains "longstring\0\0\0\0\0\0"
strlcpy(s, "foo", sizeof(s));
// s actually contains "foo\0string\0\0\0\0\0\0"
```

**Comment by stappersg** [comment]
It is _github.com_ being _github.com_, some how is
afbeelding

not in the original _merge commit message_
afbeelding

Please make commit message of 'Merge pull request https://github.com/radvd-project/radvd/pull/199 from robbat2:rjohnson/cleanups' having all **six** commits

> gitignore:_skip_newer_build/autoconf_artifacts_&_patch_droppings
> 
> Use_strlcpy_for_-Wstringop-truncation
> GCC8 added a warning for string truncation, that fires on our interface
> name handling.
> 
> Use strlcpy for handling interface names, similar to how iproute2 solved
> it.
> 
> test:_cleanup_testcases_for_-Wfree-nonheap-object
> 
> autotools:_handle_non-check_setup
> 
> radvd:_cleanups
> autotools: improve handling when libcheck is absent
> gcc-warning fixes: -Wfree-nonheap-object, -Wstringop-truncation
> gitignore: ignore a few more files
> 
> test:_include_libbsd-dev_packages

**Comment by robbat2** [comment]
We don't do squash-merge that destroys individual commit messages anyway, but I've updated it for you; also tweaked the libbsd deps a bit.

**Review by stappersg — APPROVED** [review]
Merge and move on.

**Comment by stappersg** [comment]
Not blocked my ...

afbeelding

**Pull request merged**